FEATURE: Retrieve active language dimension from TYPO3CR settings

gerhard-boden opened the pull request:
Use the TYPO3CR settings to retrieve the current language dimension within the SEO plugin. 
Example:

```
 TYPO3CR:
    languageDimension: 'language'
    contentDimensions:
      'language':
        label: 'Neos.Demo:Main:contentDimensions.language'
        icon: 'icon-language'
        ...
```

The fallback will be `language` so this is non breaking.

dlubitz commented:
What about th idea to make it more generic?
Like:
```dimensionTypes:
  language: language
  country: country```
Maybe a bit overconfigured, but cleaner than `dimensionTypeDimension: dimensionType`

gerhard-boden commented:
@dlubitz thanks, I like that!
I rebased this PR and since it already contained your bugfix I adapted it to the new way of retrieving the language dimension, so we have all in one place (easier to review).

I also added the new property to the documentation.

dlubitz commented on Resources/Private/TypoScript/Root.ts2:
```diff
@@ -65,12 +67,11 @@ prototype(TYPO3.Neos:Page) {
-			localeToLanguage = ${String.replace((item.preset ? item.preset.values[0] : item.dimensions[dimension][0]), '_', '-')}
+			localeToLanguage = ${String.replace((item.preset ? item.preset.values[0] : item.dimensions[languageDimension][0]), '_', '-')}
 
-			dimension = 'language'
```
Correct me if I'm wrong, but I think it's necessary to set the `dimension` here, because `TYPO3.Neos:DimensionMenu` needs it internaly. 
So it should be `dimension = languageDimension`.

gerhard-boden commented:
Hi @dlubitz,

thanks for your speedy review! 
Good catch! As long as there is only one content dimension present, this does not do any harm. But this setting is actually used by the to filter the dimensions. I thought this was a leftover from the earlier implementation.

gerhard-boden commented:
@kdambekalns Could you please also have a look at this when you find the time? It's basically the solution we discussed here: https://github.com/neos/neos-seo/pull/25#issuecomment-252101172

gerhard-boden commented:
@aertmann Thanks for your review!

> looks good by reading.. the default is defined twice, but that should be okay

Could you please elobarete on where you see the definition twice?
